woocommerce/woocommerce
# Analyzer: Surface hook descriptions

### All Submissions:






### Changes proposed in this Pull Request:

Add hook descriptions to Analyzer's CLI and Github output, including when hooks are modified. Also, fail when a hook is introduced without a docBlock.

Closes https://github.com/woocommerce/woocommerce/issues/33320

### How to test the changes in this Pull Request:

1. Create a new branch from this one, `test-add/analyzer-hook-desc`
2. Add a new properly commented hook

```php
/**
* Test hook that does a bunch of stuff. Here is another sentence.
*
* @since 6.7.0
*/
do_action( 'woocommerce_test_hook' );
```
3. Modify any existing hook by updating the `@since` declaration.
4. Commit changes and run the Analyzer, `./tools/code-analyzer/bin/dev analyzer test-add/analyzer-hook-desc`
5. See the CLI output contain the hook descriptions as well as appropriate messaging around updated v. introduced.

```
## HOOKS:
---------------------------------------------------
FILE: /plugins/woocommerce/includes/admin/class-wc-admin-addons.php
---------------------------------------------------
HOOK: woocommerce_test_hook: Test hook that does a bunch of stuff. Here is another sentence.
---------------------------------------------------
 NOTICE | New action found | 'woocommerce_test_hook' introduced in 6.7.0
---------------------------------------------------
FILE: /plugins/woocommerce/includes/wc-order-functions.php
---------------------------------------------------
HOOK: woocommerce_create_refund: Action hook to adjust refund before save.
---------------------------------------------------
 NOTICE | Updated action found | 'woocommerce_create_refund' updated in 6.7.0
---------------------------------------------------
```
6. Optionally, push to GH and create a test PR to see how your notices look. Otherwise, take a look at mine https://github.com/woocommerce/woocommerce/pull/33446#issuecomment-1157182669
7. Remove the description you created in the hook for step 2, commit and rerun the Analyzer, `./tools/code-analyzer/bin/dev analyzer test-add/analyzer-hook-desc`.
8. See the Analyzer error out with an appropriate error message about adding a description.
9. Now completely remove the docBlock altogether and rerun the Analyzer. Previously, this would not have been picked up and now it should also error out with an appropriate error message.

### Other information:




### FOR PR REVIEWER ONLY:
